### DISCUSSION TITLE
Is there a way to add an existing file to Gallary?

### DISCUSSION THREAD

**Discussion by ngotandatnnst** [natsuk4ze/gal#128] [Q&A] [answered]
Your plugin is clearer and better same plugins I have used

But I finding a plugin which allow me to just add a existing file (image/video) to gallary without copy and paste to gallary

I have read your code at function writeData in /main/android/src/main/java/studio/midoridesign/gal/GalPlugin.java file. It like copy and paste an existing file to gallary. It will make a file duplicated to 2, one at original path and one at new gallary path. But your plugin can do better by allow dev just tell that "add this filepath to gallary for end user can find it in Gallary app without move or copy it"

If I understand wrong, please correct me. I've only been learning Flutter for a few months

**Comment by natsuk4ze** [comment]
Thanks for starting the discussion @ngotandatnnst 🤝
Please remember to "Mark as Answered" when you resolve the issue.

Are you saying that you want to use `put-` on Android and have the gallery app scan directly the path you pass as an argument?

Basically, when we save media in a gallery, the theory is to put the files in the "Pictures" or "Movies" directory.

We could probably add them directly from the path, but if the original file is deleted for some reason, it will likely be removed from the gallery as well. That may be a unintended behavior for the developer. For example,
When the app is deleted by end user, if you had added the file to the gallery with the path obtained by `getTemporaryDirectory()` etc., the media will be removed from the gallery as well.
*If you have a better way, please let me know.

Or We could remove the file with the path received in the argument and leave only the file to be displayed in the gallery, but I think this should be removed by the developer at any given time.

> But I finding a plugin which allow me to just add an existing file (image/video) to gallary

Can you give me a more detailed use case for this? We may be able to provide support.

**Reply by ngotandatnnst** [reply]
Thanks for the quick response. 

My developing app sometime will download a huge video file up to some GB or maybe it could up to 10 GB. When download complete and gal plugin help me copy that downloaded file to Gallery, it take more resource (like power, time, storage - think about a rare use case when the phone is left 15 GB and if we need to download 10 GB video file, we will need 20 GB left to complete all actions include downloading -> copying to Gallery -> deleting original downloaded file).

It will be better if this gal plugin allows dev to decide copy to Gallery or just scan file path to Gallery. Or somehow this plugin can allow dev to save incoming stream (hasn't completed yet) direct to Gallery. Maybe by using: IOSink sink = gal.createVideosFile(fileName), and I do whatever I want with that sink object like sink.add(newBytes) in response.stream.listen. By that way, we can save double effort by bypassing copying files.

Forgive me if i'm wrong somewhere. I'm new to Flutter and it's plugins

**Comment by natsuk4ze** [comment, marked as answer]
Thank you for providing the detailed use case @ngotandatnnst !

I still think that even in that case, we should not directly scan the file in the received path and save it to the gallery. The downloaded files are probably stored in a cache directory, and I'm not sure when that would be deleted. Maybe it will be deleted just by restarting the app, and thereby unintentionally deleted from the gallery app as well.

But it certainly makes sense to have a `Stream` or some other option to save large files to the gallery.
Please consider submitting a "Feature request" issue after adding "Mark as answer."

Thank you.
